### DISCUSSION TITLE
Need for speed

### DISCUSSION THREAD

**Discussion by SebNeerman** [plausible/analytics#937] [Anything Else]
I've been doing a lot performance optimisation and speed tests recently. I have Plausible configured on a subdomain, set to async defer and not proxied (grey cloud) by CloudFlare. And I have set the subdomain to rel=preconnect in order to perform the DNS lookup, connection and SSL handshake. But I've noticed a couple of things. 

1. The preconnect (DNS, connect, SSL) sticks out like a sore thumb in the waterfall. (See screenshot)
2. A few of the online speed test tools (Insights and WebPageTest) whine that the script it's not on a CDN.

I wonder if there's anything that can be done about this. I could host the plausible.js on the main domain to take advantage of our CDN. But I'm wary that if you guys modify it further down the line I'd have to change it as well. Moreover there are a few lines in the script with relative paths to your own server and I'd have to modify these too.

Any thoughts?

Thanks!

![Screen Shot 2021-04-16 at 14 03 32](https://user-images.githubusercontent.com/153490/115024453-67f5bf00-9ec0-11eb-93b2-099491bb139f.png)

**Comment by Vigasaurus** [comment]
You definitely could host the file yourself and take advantage of your CDN, there's really only one spot that you'd need to ensure the right URL string is used - but if you download the file from the current link you're serving, it'll be correct inherently (nothing is really relative, per se). As for working through changes, odds are any future updates to the script will stay fully backwards compatible, only with new data types being excluded of course - so your existing data collection will stay intact over time and you'd only need to update the script for larger/more impactful changes as you deem appropriate. 

Past that, requesting the script on a custom domain will always have a bit of a long preconnect phase only because the CNAME resolution is notoriously a little bit slower than other verbs. 

If I remember correctly though, there were plans to transition the custom domain solution to a more powerful setup behind a better CDN itself, so maybe you could be on the lookout for that down the line too.

**Reply by SebNeerman** [reply]
Brilliant. Much appreciate the thoughtful reply.
